Sources/Datadog/Logger.swift

```diff
@@ -460,49 +364,104 @@ public class Logger {
-            )
-        }
+            let consoleLogger: ConsoleLogger? = {
+                guard let consoleLogFormat = consoleLogFormat else {
+                    return nil
+                }
 
-        private func resolveOuput() -> LogOutput? {
-            switch (useCoreOutput, consoleLogFormat) {
-            case let (true, format?):
-                return CombinedLogOutput(
-                    combine: [
-                        LogConsoleOutput(format: format, timeZone: .current),
-                        LoggingWithRUMErrorsIntegration()
-                    ]
+                let configuration = ConsoleLogger.Configuration(
+                    timeZone: .current,
+                    format: consoleLogFormat
                 )
-            case (true, nil):
-                return LoggingWithRUMErrorsIntegration()
-            case let (false, format?):
-                return LogConsoleOutput(format: format, timeZone: .current)
-            case (false, nil):
-                return nil
+
+                return ConsoleLogger(
+                    configuration: configuration,
+                    dateProvider: SystemDateProvider(),
+                    printFunction: consolePrint
+                )
+            }()
+
+            switch (remoteLogger, consoleLogger) {
+            case (let remoteLogger?, nil):
+                return remoteLogger
+
+            case (nil, let consoleLogger?):
+                return consoleLogger
+
+            case (let remoteLogger?, let consoleLogger?):
+                return CombinedLogger(combinedLoggers: [remoteLogger, consoleLogger])
+
+            case (nil, nil): // when user explicitly produces a no-op logger
+                return NoOpLogger()
             }
         }
     }
 }
+
+/// Combines multiple loggers together into single `Logger` interface.
+internal struct CombinedLogger: LoggerProtocol {
+    let combinedLoggers: [LoggerProtocol]
+
+    func log(level: LogLevel, message: String, error: Error?, attributes: [String: Encodable]?) {
+        combinedLoggers.forEach { $0.log(level: level, message: message, error: error, attributes: attributes) }
+    }
+
+    func addAttribute(forKey key: AttributeKey, value: AttributeValue) {
+        combinedLoggers.forEach { $0.addAttribute(forKey: key, value: value) }
+    }
+
+    func removeAttribute(forKey key: AttributeKey) {
+        combinedLoggers.forEach { $0.removeAttribute(forKey: key) }
+    }
+
+    func addTag(withKey key: String, value: String) {
+        combinedLoggers.forEach { $0.addTag(withKey: key, value: value) }
+    }
+
+    func removeTag(withKey key: String) {
+        combinedLoggers.forEach { $0.removeTag(withKey: key) }
+    }
+
+    func add(tag: String) {
+        combinedLoggers.forEach { $0.add(tag: tag) }
+    }
+
+    func remove(tag: String) {
+        combinedLoggers.forEach { $0.remove(tag: tag) }
+    }
+}
+
+internal struct NoOpLogger: LoggerProtocol {
```

Comment:
```suggestion
internal struct NOPLogger: LoggerProtocol {
```

cf. https://github.com/DataDog/dd-sdk-ios/pull/921#discussion_r917723968